src/server/luaentity_sao.cpp

```diff
@@ -200,16 +259,23 @@ void LuaEntitySAO::step(float dtime, bool send_recommended)
 	if(!isAttached())
 	{
 		// TODO: force send when acceleration changes enough?
-		float minchange = 0.2*BS;
-		if(m_last_sent_position_timer > 1.0){
-			minchange = 0.01*BS;
-		} else if(m_last_sent_position_timer > 0.2){
-			minchange = 0.05*BS;
-		}
-		float move_d = m_base_position.getDistanceFrom(m_last_sent_position);
-		move_d += m_last_sent_move_precision;
+		float minchange = 0.2 * BS;
+		if (m_last_sent_position_timer > 1.0)
+			minchange = 0.01 * BS;
+		else if (m_last_sent_position_timer > 0.2)
+			minchange = 0.05 * BS;
+
+		float pos_d = m_base_position.getDistanceFrom(m_last_sent_position);
+		pos_d += m_last_sent_move_precision;
 		float vel_d = m_velocity.getDistanceFrom(m_last_sent_velocity);
-		if (move_d > minchange || vel_d > minchange ||
+
+		// TODO: Find a better way to overcome buggy client-side
+		// synchronization with velocity changes since the client
+		// does not correctly predict entity-player collisions.
+		bool force_send = m_last_sent_position_timer > 0.5 &&
+			m_velocity.getLength() > 0.05 && m_prop.collideWithObjects;
```

Comment:
Good idea, I added a check for `velocity_lock`. Relying on `vel_d` doesn't solve the problem because the client keeps moving the object after it collides with another object, so it needs to be continuously corrected. The  ideal solution is to improve the client-side prediction.